File: 1039-find-the-town-judge/find-the-town-judge.cpp
class Solution {
public:
    int findJudge(int n, vector<vector<int>>& trust) {
        int cnt = 0; 
        int ans =  -1; 
        if(n == 1) return 1;
        unordered_map<int,int> u;
        unordered_map<int,int> v;
        for(int i = 0 ;i<trust.size() ;i++){
            u[trust[i][1]]++;
            v[trust[i][0]]++;
        }

        for(auto it : u){
            if(it.second == n-1 && (v[it.first]==0)){
                ans = it.first;
                cnt++;
            }
        }
        if(cnt>1) return -1;
        return ans;
    }
};

File: 1039-find-the-town-judge/find-the-town-judge.py
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        ins = [0]*(n+1)
        out = [0] * (n+1)
        for a in trust:
            out[a[0]]+=1
            ins[a[1]]+=1
        for i in range(1,n+1):
            if ins[i] == n-1 and out[i] == 0:
                return i
        return -1